### ISSUE TITLE
Triton provider keeps waiting for failed provision job to complete

### ISSUE THREAD

**Issue by StanYago** [hashicorp/terraform#13447]
### Terraform Version
Terraform v0.9.2

### Affected Resource(s)
- triton_machine

### Terraform Configuration Files
```hcl
resource "triton_machine" "test-kafka" {
  name    = "test-kafka"
  package = "kafka"
  image   = "${var.base64-lts}"
}
```

### Expected Behavior
Triton job failed.  Terraform should have been notified of this failure. 

### Actual Behavior
Terraform kept waiting for job to complete for 10m waiting for triton_machine to finish.

### Steps to Reproduce
1. Create a package with previously undefined trait in Triton
2. Specify the package in terraform
3. `terraform apply`

### Important Factoids
Potentially any failure in provisioner is not reflected back in terraform.

**Comment by jen20** [comment]
Hi @StanYago! Thanks for opening an issue. Can you confirm that by "provisioning" here you mean the act of creating the machine in Triton as opposed to Terraform provisioning? It's likely that this is fixed in Terraform 0.9.4 as we are using a newer underlying SDK which has fewer issues around failures, but I will try to reproduce this.

**Comment by StanYago** [comment]
Hi @jen20, yes by provisioning I meant creating a VM in Triton via terraform.  I currently don't have access to my private cloud to validate if this is fixed in Terraform 0.9.4.  However, as far as error handling is concerned, I noticed that if I specify a non-resolvable URL, no error is returned by terraform and it keeps waiting indefinitely for a VM to be created.

**Comment by StanYago** [comment]
I am still able to reproduce the original issue with Terraform v0.9.6.  When terraform initiated triton provision job fails, terraform does not receive the error and waits for 10 minutes before failing.  
When Triton provision jobs fails, Triton still allocates UUID and returns that to terraform which terraform marks as tainted.  On the next run terraform will attempt to destroy tainted VM that is already marked as failed in Triton.  In this case no job is created in triton, but terraform shows "Still destroying... (ID: ef54a447-...-...-...   )" for the next 10 minutes.
